src/nvim/syntax.c

```diff
@@ -5211,11 +5203,11 @@ static void syn_cmd_sync(exarg_T *eap, int syncing)
       n = getdigits_long(&arg_end, false, 0);
```

Comment:
This `n` can probably be changed to `int32_t` or `linenr_T` and use `getdigits_int32` here.